packages/ui/src/Checkbox.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { AriaAttributes, ChangeEvent, FC, FocusEvent, useRef } from 'react'
+import styles from './Checkbox.module.scss'
+import { Check, Minus } from 'react-feather'
+import { DataTestProp } from '@hazelcast/helpers'
+import classNames from 'classnames'
+import { Error, errorId } from './Error'
+import { Help, helpTooltipId } from './Help'
+import { v4 as uuid } from 'uuid'
+
+type CheckboxCoreProps = {
+  name: string
+  value?: string
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void
+  error?: string
+  checked?: boolean
+}
+
+export type CheckboxExtraProps = {
+  label: string
+  helperText?: string
+  indeterminate?: boolean
+  disabled?: boolean
+  required?: boolean
+  className?: string
+  classNameLabel?: string
+}
+
+type CheckboxProps = CheckboxCoreProps & CheckboxExtraProps & DataTestProp
+
+/**
+ * ### Purpose
+ * Forms require input from users. If you need information that can be represented as boolean, use input with type 'checkbox'.
+ * Checkbox fields accept boolean types of data. And help and error guidance to ensure they know what to enter.
+ *
+ *
+ * ### General Info
+ * - It's important to realise that we don't set checkbox value, but state (on/off). That is the main difference from other input types.
+ * - They can have a special indeterminate state, that represents a '3rd' value, usually used in tree structures, etc.
+ */
+export const Checkbox: FC<CheckboxProps> = ({
+  checked,
+  name,
+  onChange,
+  onBlur,
+  className,
+  classNameLabel,
+  value,
+  indeterminate = false,
+  label,
+  helperText,
+  error,
+  disabled = false,
+  required,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null)
+  const idRef = useRef(uuid())
+
+  let ariaChecked: AriaAttributes['aria-checked'] = checked ? 'true' : 'false'
+  if (indeterminate) {
+    ariaChecked = 'mixed'
+  }
+
+  return (
+    <div className={className}>
+      {/* 
+        We can only style forward elements based on input state (with ~ or +), has() is not supported yet.
+        That's why we need to explicitly pass error/checked/disabled classes to the wrapper element.
+      */}
+      <label
+        className={classNames(
+          styles.wrapper,
+          {
+            [styles.error]: !!error,
+            [styles.checked]: checked,
+            [styles.disabled]: disabled,
+          },
+          classNameLabel,
+        )}
+        htmlFor={idRef.current}
+      >
+        <span className={styles.name}>{label}</span>
+        <input
+          type="checkbox"
+          ref={inputRef}
+          id={idRef.current}
+          name={name}
+          checked={checked}
```

Comment:
To make the checkbox controlled even when we pass `checked` as `undefined`
```suggestion
          checked={!!checked}
```